app/src/main/contrib/sources/ParametersTokenSource.ts

```diff
@@ -0,0 +1,26 @@
+import SourceModel = AltJS.SourceModel;
+import {settings} from "../../shared/Settings";
+import {modelParameterActions} from "../actions/ModelParameterActions";
+import {ResponsibilityState} from "../stores/ResponsibilityStore";
+import {Source} from "../../shared/sources/Source";
+
+export class ModelParametersTokenSource extends Source<ResponsibilityState> {
+
+    _fetchOneTimeParametersToken: () => SourceModel<string>;
+
+    constructor() {
+        super();
+        this._fetchOneTimeParametersToken = () => this.doFetch(state => {
+            let queryString = "";
+            if (state.redirectPath && state.redirectPath.length > 0) {
+                queryString = "?redirectUrl=" + encodeURI(settings.montaguUrl() + state.redirectPath);
+            }
```

Comment:
The redirect path code is duplicated from elsewhere - could you refactor to this to be more DRY?